Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -49,6 +98,19 @@ class PreplannedMapModel: ObservableObject, Identifiable {
         }
     }
     
+    /// Sets the model download preplanned offline map job if the job is in progress.
+    private func setDownloadJob() {
+        for case let preplannedJob as DownloadPreplannedOfflineMapJob in JobManager.shared.jobs {
+            if preplannedJob.downloadDirectoryURL.deletingPathExtension().lastPathComponent == preplannedMapAreaID {
+                job = preplannedJob
+                status = .downloading
+                Task {
```

Comment:
This task should probably be detached, and self captured weakly.

But it should also probably be broken out into a function.